gpu/assets.go

```diff
@@ -32,10 +32,11 @@ func Cleanup() {
 	lock.Lock()
 	defer lock.Unlock()
 	if payloadsDir != "" {
-		slog.Debug("cleaning up", "dir", payloadsDir)
-		err := os.RemoveAll(payloadsDir)
+		tmpDir := filepath.Clean(filepath.Join(payloadsDir, ".."))
```

Comment:
What is above the payloads dir? Can we assume we have write access?

Reply:
I'll add some comments to help clarify what's going on.  Line 26 adds a new sub directory above the tmpDir so that we can disambiguate between rocm runner and the rocm sym link.